service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandlerTest.java

```diff
@@ -600,6 +614,80 @@ void shouldNotifyAdminOfPlacementApplicationUpload() throws Exception {
+            preferences)).willReturn(expectedParameters);
+
+        given(partyAddedToCaseContentProvider.getPartyAddedToCaseNotificationTemplate(preferences))
+            .willReturn(PARTY_ADDED_TO_CASE_BY_EMAIL_NOTIFICATION_TEMPLATE);
+
+        notificationHandler.sendNotificationToPartiesAddedToCase(new PartyAddedToCaseEvent(callbackRequest(), AUTH_TOKEN, USER_ID, representatives));
+
+        verify(notificationClient, times(1)).sendEmail(
+            eq(PARTY_ADDED_TO_CASE_BY_EMAIL_NOTIFICATION_TEMPLATE), eq(PARTY_ADDED_TO_CASE_BY_EMAIL_ADDRESS),
+            eq(expectedParameters), eq("12345"));
+    }
+
+    @Test
+    void shouldSendNotificationToPartiesWhenAddedToCaseThroughDigitalService() throws IOException, NotificationClientException {
+        final Map<String, Object> expectedParameters = ImmutableMap.<String, Object>builder()
+            .put("firstRespondentLastName", "Moley")
+            .put("familyManCaseNumber", "123")
+            .put("reference", "12345")
+            .build();
+
+        List<Element<Representative>> representatives = new ArrayList<>();
+        Element<Representative> representative = Element.<Representative>builder()
+            .value(Representative.builder()
+                .email("damian@swansea.gov.uk")
+                .servingPreferences(DIGITAL_SERVICE).build()).build();
+        representatives.add(representative);
+
+        RepresentativeServingPreferences preferences = DIGITAL_SERVICE;
+
+        given(partyAddedToCaseContentProvider.getPartyAddedToCaseNotificationParameters(callbackRequest().getCaseDetails(),
+            preferences)).willReturn(expectedParameters);
+
+        given(partyAddedToCaseContentProvider.getPartyAddedToCaseNotificationTemplate(preferences))
+            .willReturn(PARTY_ADDED_TO_CASE_THROUGH_DIGITAL_SERVICE_NOTIFICATION_TEMPLATE);
+
+        notificationHandler.sendNotificationToPartiesAddedToCase(new PartyAddedToCaseEvent(callbackRequest(), AUTH_TOKEN, USER_ID, representatives));
+
+        verify(notificationClient, times(1)).sendEmail(
+            eq(PARTY_ADDED_TO_CASE_THROUGH_DIGITAL_SERVICE_NOTIFICATION_TEMPLATE), eq(PARTY_ADDED_TO_CASE_THROUGH_DIGITAL_SERVICE_EMAIL),
+            eq(expectedParameters), eq("12345"));
+    }
+
+    @Test
+    void shouldNotSendNotificationToPartiesWhenAddedToCaseThroughPost() throws IOException, NotificationClientException {
+        List<Element<Representative>> representatives = new ArrayList<>();
+        Element<Representative> representative = Element.<Representative>builder()
+            .value(Representative.builder()
+                .email("damian@swansea.gov.uk")
+                .servingPreferences(POST).build()).build();
+        representatives.add(representative);
+
+        notificationHandler.sendNotificationToPartiesAddedToCase(new PartyAddedToCaseEvent(callbackRequest(), AUTH_TOKEN, USER_ID, representatives));
+
+        verify(notificationClient, times(0)).sendEmail(null, null, null, null);
```

Comment:
You can do `never()` instead of `times(0)`

Reply:
can we use `any...` in place of `nulls`?

Reply:
ah yes i didn't like the nulls either :)